composer.json

```diff
@@ -20,15 +20,16 @@
     },
     "require": {
         "php": ">=5.3.3",
-        "ezyang/htmlpurifier": "^4.7",
         "kub-at/php-simple-html-dom-parser": "^1.7"
     },
     "require-dev": {
         "phpunit/phpunit": "~5.0",
-        "doctrine/cache": "~1.0"
+        "doctrine/cache": "~1.0",
+        "ezyang/htmlpurifier": "^4.7"
     },
     "suggest": {
-        "doctrine/cache": "Used for caching the calculated diffs using a Doctrine Cache Provider"
+        "doctrine/cache": "Used for caching the calculated diffs using a Doctrine Cache Provider",
+        "ezyang/htmlpurifier": "Used to purify inputted HTML, which is used by default. Either disable it, or install this bundle."
```

Comment:
I disagree with this, 99% of the time you will need this vendor, since non purified HTML will not diff correctly.

We rely heavily on clean input, since allot of the manipulation is done using simple string manipulation at the moment.

I don't think this is a bad feature request, but I think it should be an option for people that `know what they are doing`, meaning, they run HTMLPurifier themselves always before requesting a diff.

Reply:
I agree with your point that you need cleaned input for a correct diff, but as stated, sometimes you already have a cleaned input. And you might have done that with another library, or maybe purify in this context is not necessary as you already generate clean HTML (which is my use case). 

Enforcing to install a specific vendor is a waste of resources in those cases: and if you don't explicitly disable the purifier, you will be presented with an error which suggest installing it.

Reply:
Ah yes, but the real problem at this moment is that even clean HTML will not diff correctly.

I will give an example.

```
<html>
 <body>
  <p class='test'>hi</a>
 </body>
</html>
```

This is parseable HTML, and could be considered clean.

However HTMLPurifier will change the single qoutes from the class attribute to double quotes, even though in the w3c spec both are allowed.

I have seen bugs where the output from the library will return this `class="'test diffmod'`. Of course, this is a bug in the library and should be fixed, but that is a bit outside of the scope of this conversation.

My point is that we rely on the purifier, its a pretty integral part of the library, this is not intentional, but that's the way it is, for now, so we will have to deal with it.

Again I am not against a config flag that says `Don't purify the input` for performance reasons.

Reply:
Ok, fair enough. If I move the purifier back to the composer required section, would that be enough to get this merged?

Reply:
Sounds like a plan, just some small remarks, see the other two comment.